crates/bevy_ecs/src/system/system.rs

```diff
@@ -82,6 +82,9 @@ pub trait System: Send + Sync + 'static {
     unsafe fn run_unsafe(&mut self, input: SystemIn<'_, Self>, world: UnsafeWorldCell)
         -> Self::Out;
 
+    /// Refresh the inner pointer based on the latest hot patch jump table
+    fn refresh_hotpatch(&mut self);
```

Comment:
```suggestion
    /// Refresh the inner pointer based on the latest hot patch jump table
    #[cfg(feature = "hotpatching")]
    fn refresh_hotpatch(&mut self) {}
```
If we provide a default implementation (no-op), we can safely feature-gate this method while keeping features purely additive. I think this is desirable, as the inner implementation for a user's `refresh_hotpatch` might only make sense when the feature is enabled, potentially forcing them to add their own gate within the body of the function.

Alternatively, could this function be modified to be more general? E.g., `refresh`?

Reply:
I didn't want to cfg-gate all the functions in all the implementation of that trait, but it makes sense...

Before trying to find a more general version, let's wait for systems as entities

Reply:
cfg- gated them 👍